pkg/node/node.go

```diff
@@ -278,11 +279,25 @@ func NewBee(
 		return nil, fmt.Errorf("compute overlay address: %w", err)
 	}
 
-	if nonceExists && o.TargetNeighborhood != "" {
-		logger.Warning("an overlay has already been created before, skipping targeting the selected neighborhood")
-	}
+	if o.TargetNeighborhood != "" {
+		if nonceExists {
+			logger.Info("Override nonce %d and clean state for neighborhood %s", nonce, o.TargetNeighborhood)
+			logger.Warning("you have another 10 seconds to change your mind and kill this process with CTRL-C...")
+			time.Sleep(10 * time.Second)
 
-	if !nonceExists {
+			const (
+				localstore = "localstore"
+				kademlia   = "kademlia-metrics"
+				statestore = "statestore"
+			)
+			dirsToNuke := []string{localstore, kademlia, statestore}
+			for _, dir := range dirsToNuke {
+				err = removeContent(filepath.Join(o.DataDir, dir))
+				if err != nil {
+					return nil, fmt.Errorf("delete %s: %w", dir, err)
+				}
+			}
+		}
 		// mine the overlay
 		targetNeighborhood := o.TargetNeighborhood
 		if o.TargetNeighborhood == "" && o.NeighborhoodSuggester != "" {
```

Comment:
for the suggester to run. the target neighborhood has to be "",  but this check is inside the `if o.TargetNeighborhood != "" {` check on line 283, no?

Reply:
correct, good catch! : ) move outside of the block